homeassistant/components/devolo_home_control/climate.py

```diff
@@ -47,6 +48,22 @@ class DevoloClimateDeviceEntity(DevoloMultiLevelSwitchDeviceEntity, ClimateEntit
 
     @property
     def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        if hasattr(self._device_instance, "multi_level_sensor_property"):
+            return next(
+                (
+                    multi_level_sensor.value
+                    for element_uid, multi_level_sensor in self._device_instance.multi_level_sensor_property.items()
+                    if multi_level_sensor.sensor_type == "temperature"
+                ),
+                self.target_temperature,
+            )
+
+        # For devices, that don't report their current temperature, return target temperature.
```

Comment:
If a device doesn't report current temperature we should return None in that case.

Reply:
Ok, I changed it, but from my point of view, the UI now does not look very nice.

![grafik](https://user-images.githubusercontent.com/20171515/95502428-1c5ed780-09aa-11eb-86b0-2656d65a0712.png)

Reply:
That's an improvement to be made in the frontend. Current temperature is optional so the frontend should handle that.